pkg/pgmodel/ingestor/buffer.go

```diff
@@ -0,0 +1,88 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package ingestor
+
+import (
+	"sync"
+
+	"github.com/timescale/promscale/pkg/pgmodel/common/schema"
+	"github.com/timescale/promscale/pkg/pgmodel/utils"
+)
+
+const (
+	// maximum number of insertDataRequests that should be buffered before the
+	// insertHandler flushes to the next layer. We don't want too many as this
+	// increases the number of lost writes if the connector dies. This number
+	// was chosen arbitrarily.
+	flushSize                = 2000
+	getCreateMetricsTableSQL = "SELECT table_name FROM " + schema.Catalog + ".get_or_create_metric_table_name($1)"
+	finalizeMetricCreation   = "CALL " + schema.Catalog + ".finalize_metric_creation()"
+	getSeriesIDForLabelSQL   = "SELECT * FROM " + schema.Catalog + ".get_or_create_series_id_for_kv_array($1, $2, $3)"
+	getEpochSQL              = "SELECT current_epoch FROM " + schema.Catalog + ".ids_epoch LIMIT 1"
+	maxCopyRequestsPerTxn    = 100
+)
+
+// epoch for the ID caches, -1 means that the epoch was not set
```

Comment:
Do you think it's good to have a `const EpochNotSet = Epoch(-1)` so we can use it as a const throughout the code?

Reply:
I am not sure. The changes as such are huge. Maybe, a follow on PR (by me or someone else who has more used to epoch code) for this and the comment below for global state?

Reply:
👍